fades/helpers.py

```diff
@@ -208,3 +211,40 @@ def check_pypi_updates(dependencies):
 
     dependencies["pypi"] = dependencies_up_to_date
     return dependencies
+
+
+def _pypi_head_package(dependency):
+    """Hit pypi with a http HEAD to check if pkg_name exists."""
+    if dependency.specs:
+        _, version = dependency.specs[0]
+        url = BASE_PYPI_URL_WITH_VERSION.format(name=dependency.project_name, version=version)
+    else:
+        url = BASE_PYPI_URL.format(name=dependency.project_name)
+    logger.debug("Doing HEAD requests agains %s", url)
+    req = request.Request(url, method='HEAD')
+    try:
+        response = request.urlopen(req)
+    except HTTPError as http_error:
+        if http_error.code == HTTP_STATUS_NOT_FOUND:
+            return False
+        else:
+            raise http_error
+    if response.status == HTTP_STATUS_OK:
+        logger.debug("%r exists in Pypi.", dependency)
+        return True
+
```

Comment:
I wonder... what if there was not an http_error, but status is not OK, what would or should happen?

IOW, there is a missing "else" here.

Maybe we could consider that if we're without an exception, it's OK, and just return True? (without checking equality against HTTP_STATAUS_OK)

Reply:
I added a "else" and a log in warning.